generators/app/gulpfile.js

```diff
@@ -11,72 +11,90 @@
 'use strict';
 
 const path = require('path');
+
+const del = require('del');
 const gulp = require('gulp');
 const gulpif = require('gulp-if');
+const imagemin = require('gulp-imagemin');
+const mergeStream = require('merge-stream');
+const polymerBuild = require('polymer-build');
+
+const swPrecacheConfigPath = path.join(process.cwd(), 'sw-precache-config.js');
+const swPrecacheConfig = require(swPrecacheConfigPath);
+const polymerJsonPath = path.join(process.cwd(), 'polymer.json');
+const polymerJson = require(polymerJsonPath);
+const polymerProject = new polymerBuild.PolymerProject(polymerJson);
+const buildDirectory = 'build';
+
+/**
+ * Waits for the given ReadableStream
+ */
+function waitFor(stream) {
+  return new Promise((resolve, reject) => {
+    stream.on('end', resolve);
+    stream.on('error', reject);
+  });
+}
 
-// Got problems? Try logging 'em
-// const logging = require('plylog');
-// logging.setVerbose();
+function build() {
+  return new Promise((resolve, reject) => { // eslint-disable-line no-unused-vars
+    // Okay, so first thing we do is clear the build directory
+    console.log(`Deleting ${buildDirectory} directory...`);
+    del([buildDirectory])
+      .then(() => {
+        // Okay, now lets get your source files
+        let sourcesStream = polymerProject.sources()
+          // Oh, well do you want to minify stuff? Go for it!
+          // Here's how splitHtml & gulpif work
+          .pipe(polymerProject.splitHtml())
+          // .pipe(gulpif(/\.js$/, uglify()))
+          // .pipe(gulpif(/\.css$/, cssSlam()))
+          // .pipe(gulpif(/\.html$/, htmlMinifier()))
+          .pipe(gulpif(/\.(png|gif|jpg|svg)$/, imagemin({
```

Comment:
Shouldn't this be:
```js
.pipe(gulpif(/\.(png|gif|jpg|svg)$/, imagemin([
  imagemin.gifsicle({interlaced: true}),
  imagemin.jpegtran({progressive: true}),
  imagemin.optipng(),
  imagemin.svgo()
])))
```
As stated for imagemin 3.0+ on their [3.0.0 Release Notes](https://github.com/sindresorhus/gulp-imagemin/releases/tag/v3.0.0)
Note: Though they state `mozjpeg`, the exported function actually seems to be `jpegtran`.